src/DynamoCore/Library/FunctionDescriptor.cs

```diff
@@ -568,4 +569,217 @@ private string GetRootCategory()
             return string.IsNullOrEmpty(Namespace) ? filename : filename + "." + Namespace;
         }
     }
+
+    internal class ReflectionFunctionDescriptor : FunctionDescriptor
+    {
+        private readonly LibraryCustomization libraryCustomization;
+        private readonly Assembly reflectionAsm;
+
+        public new string Category
```

Comment:
Any chance `FunctionDescriptor` can be flaged as `virtual`? That way here you could do `public override string Category` instead. 
Only issue with the `new string Category` is when casting the `ReflectionFunctionDescriptor as FunctionDescriptor` in that case `FunctionDescriptor.Category` will be used instead.

Reply:
Also, if the intent is to defer the `Category` computation to when is actually needed, consider using `Lazy<string> category` so it only executes the `get` code once.
```csharp
    private string Lazy<string> category = new Lazy<string>(() => {
        // Code for getting the category goes here  
    });
   
    public override string Category => category.Value;
```

Reply:
> Only issue with the new string Category is when casting the ReflectionFunctionDescriptor as FunctionDescriptor in that case FunctionDescriptor.Category will be used instead.

Yea thats actually a pretty good point! 
Not completely sure if that would be considered an API-breaking change though. Ill bring it up with the dynamo team.

https://docs.microsoft.com/en-us/dotnet/core/compatibility/
![image](https://user-images.githubusercontent.com/13732445/119478976-1d315780-bd48-11eb-8598-91f77e06c482.png)

Reply:
Also, the need for this ReflectionFunctionDescriptor is simply that the normal `FunctionDescriptor` uses `GetCustomAttribute` which is not possible when types are loaded in reflection only, so the `ReflectionFunctionDescriptor` simply provides a different implementation that can be used with reflection only types